upload/lambdas/health_check/health_check.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime, timedelta
+import json
+import os
+
+import boto3
+import requests
+
+from upload.common.database import _run_query
+from upload.common.logging import get_logger
+from upload.common.upload_config import UploadConfig
+
+logger = get_logger(__name__)
+
+client = boto3.client('cloudwatch')
+
+
+class HealthCheck:
+    def __init__(self):
+        self.env = os.environ['DEPLOYMENT_STAGE']
+        logger.debug(f"Running a health check for {self.env}. Results will be posted in #upload-service")
+        self.webhook = UploadConfig().slack_webhook
+
+        self.checksumming_areas_count_query = "SELECT COUNT(*) FROM checksum " \
+                                              "WHERE status='CHECKSUMMING' " \
+                                              "AND created_at > CURRENT_DATE - interval '4 weeks' " \
+                                              "AND updated_at > CURRENT_TIMESTAMP - interval '2 hours'"
+        self.validation_areas_count_query = "SELECT COUNT(*) FROM validation " \
+                                            "WHERE status='VALIDATING' " \
+                                            "AND created_at > CURRENT_DATE - interval '4 weeks' " \
+                                            "AND updated_at > CURRENT_TIMESTAMP - interval '2 hours'"
+
+        self.undeleted_areas_count_query = "SELECT COUNT(*) FROM upload_area " \
+                                           "WHERE created_at > CURRENT_DATE - interval '4 weeks' " \
+                                           "AND status != 'DELETED'"
+
```

Comment:
can we add two more queries here to find "UNSCHEDULED" validation and checksum jobs created in last month but not updated in last 2 hours?

Reply:
👍 Scheduled